reposerver/repository/repository.go

```diff
@@ -564,16 +566,38 @@ func resolveReferencedSources(hasMultipleSources bool, source *v1alpha1.Applicat
 			}
 			return nil, fmt.Errorf("source referenced %q, which is not one of the available sources (%s)", refVar, strings.Join(refKeys, ", "))
 		}
-		if refSourceMapping.Chart != "" {
-			return nil, errors.New("source has a 'chart' field defined, but Helm charts are not yet not supported for 'ref' sources")
+
+		// Allow OCI ref sources with chart field, but still restrict Git ref sources with chart field
+		if refSourceMapping.Chart != "" && !refSourceMapping.Repo.IsOCI() {
+			return nil, errors.New("source has a 'chart' field defined, but Helm charts are not yet supported for Git 'ref' sources")
+		}
+
+		var normalizedRepoURL string
+		if v1alpha1.IsOCIURL(refSourceMapping.Repo.Repo) {
+			// For OCI repositories, use the URL as-is without normalization
+			normalizedRepoURL = refSourceMapping.Repo.Repo
+		} else {
+			// For Git repositories, normalize the URL
+			normalizedRepoURL = git.NormalizeGitURL(refSourceMapping.Repo.Repo)
 		}
-		normalizedRepoURL := git.NormalizeGitURL(refSourceMapping.Repo.Repo)
+
```

Comment:
I think we can just use 
```suggestion
		normalizedRepoURL := git.NormalizeGitURL(refSourceMapping.Repo.Repo)
```

This should be fine to normalize even with OCI URLs (all the logic in there is specific to Git itself and shouldn't affect OCI)